src/components/main/main-logic.tsx

```diff
@@ -1,35 +1,35 @@
-import { useCallback, useEffect, useState } from 'react';
+import { useFetchPokemonsQuery } from '@utils/api/pokemon-api';
+import { useHandleLocalStorage } from '@utils/hooks';
+import type { Pokemon } from '@utils/types';
+import { useCallback } from 'react';
 
-import useHandleLocalStorage from '../../hooks/use-handle-local-storage';
-import { PokemonService } from '../../services/pokemonService';
-import type { Pokemon } from '../../types';
-
-const MainLogic = () => {
+const BodyPageLogic = () => {
   const [searchText, setSearchText] = useHandleLocalStorage(
     'savedSearchPokemon',
     ''
   );
-  const [pokemons, setPokemons] = useState<Pokemon[]>([]);
-
-  const loadPokemons = useCallback(async () => {
-    const fetchedPokemons = await new PokemonService().fetchPokemons();
-    setPokemons(fetchedPokemons);
-  }, []);
+  const {
+    data: apiResponse,
+    isLoading,
+    isError,
+    refetch,
+  } = useFetchPokemonsQuery(20);
 
-  useEffect(() => {
-    loadPokemons();
-  }, [loadPokemons]);
+  const pokemons = apiResponse?.results || [];
```

Comment:
Насколько знаю, использование `??` сейчас считается предпочтительным - более явно показывает, что мы хотим избежать `null` или `undefined`, но не других ложных значений
```suggestion
  const pokemons = apiResponse?.results ?? [];
```